executor/executor_simple_test.go

```diff
@@ -223,20 +220,16 @@ func (s *testSuite) TestSetPwd(c *C) {
 func (s *testSuite) TestAnalyzeTable(c *C) {
 	defer testleak.AfterTest(c)()
 	tk := testkit.NewTestKit(c, s.store)
-	tk.MustExec(`ANALYZE TABLE mysql.GLOBAL_VARIABLES`)
-	ctx := tk.Se.(context.Context)
-	is := sessionctx.GetDomain(ctx).InfoSchema()
-	t, err := is.TableByName(model.NewCIStr("mysql"), model.NewCIStr("GLOBAL_VARIABLES"))
-	c.Check(err, IsNil)
-	tableID := t.Meta().ID
-
-	err = ctx.NewTxn()
-	c.Check(err, IsNil)
-	meta := meta.NewMeta(ctx.Txn())
-	tpb, err := meta.GetTableStats(tableID)
-	c.Check(err, IsNil)
-	c.Check(tpb, NotNil)
-	tStats, err := statistics.TableFromPB(t.Meta(), tpb)
-	c.Check(err, IsNil)
-	c.Check(tStats, NotNil)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec("create table t1 (a int)")
+	tk.MustExec("create index ind_a on t1 (a)")
+	tk.MustExec("insert into t1 (a) values (1)")
+	result := tk.MustQuery("explain select * from t1 where t1.a = 1")
+	rowStr := fmt.Sprintf("%s", result.Rows())
+	c.Check(strings.Split(rowStr, "{")[0], Equals, "[[IndexScan_5 ")
+	tk.MustExec("analyze table t1")
+	result = tk.MustQuery("explain select * from t1 where t1.a = 1")
+	rowStr = fmt.Sprintf("%s", result.Rows())
+	c.Check(strings.Split(rowStr, "{")[0], Equals, "[[TableScan_4 ")
```

Comment:
It is a good test case, but I don't think you will get the desired result, yet. Because currently, the logic for index columns is, somewhat, designed for pseudo cases, which in this case will always return `table.Count / 3`. After the two PR is merged, this test case will be testable. I will add it in the next PR. Always next PR...